### PR TITLE
feat: add get subscriber preference usecase

### PR DESCRIPTION
**Pull request by djabarovgeorge**

### CODE REVIEW

File: libs/dal/src/repositories/subscriber-preference/subscriber-preference.repository.ts [hunk 1]
```diff
@@ -6,4 +6,17 @@ export class SubscriberPreferenceRepository extends BaseRepository<SubscriberPre
   constructor() {
     super(SubscriberPreference, SubscriberPreferenceEntity);
   }
+
+  async findAll(
```
Comment (ainouzgali) [inline libs/dal/src/repositories/subscriber-preference/subscriber-preference.repository.ts:10]: Should we maybe use a more clear name of what we are finding? But it's not a big deal..

File: apps/api/src/app/widgets/e2e/get-subscriber-preference.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import { NotificationTemplateEntity, SubscriberRepository, SubscriberPreferenceRepository } from '@novu/dal';
+import { UserSession } from '@novu/testing';
+import axios from 'axios';
+import { expect } from 'chai';
+
+describe('GET /widget/notifications/feed', function () {
```
Comment (ainouzgali) [inline apps/api/src/app/widgets/e2e/get-subscriber-preference.e2e.ts:6]: So great you started on the tests💙

**Review by ainouzgali — APPROVED** [review]
Loks great @djabarovgeorge !

File: libs/dal/src/repositories/subscriber-preference/subscriber-preference.repository.ts (hunk 1 above)
Comment (djabarovgeorge) [reply libs/dal/src/repositories/subscriber-preference/subscriber-preference.repository.ts:10]: @ainouzgali thanks for raising it up.  The method is fetching all the subscriber's temples. What name would you suggest?

Comment (ainouzgali) [reply libs/dal/src/repositories/subscriber-preference/subscriber-preference.repository.ts:10]: maybe findSubscriberTemplates? @djabarovgeorge

File: apps/api/src/app/widgets/e2e/get-subscriber-preference.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    await session.triggerEvent(template.triggers[0].identifier, subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, subscriberId);
+
+    await session.awaitRunningJobs(template._id);
+
+    const response = await axios.get(`http://localhost:${process.env.PORT}/v1/widgets/subscriber-preference`, {
+      headers: {
+        Authorization: `Bearer ${subscriberToken}`,
+      },
+    });
```
Comment (scopsy) [inline apps/api/src/app/widgets/e2e/get-subscriber-preference.e2e.ts:51]: Do you think we can abstract the session initialize and the widget requests under the session class? I see we have quite a lot of those usages already. Maybe doing something moregeneric to abstract the subscriber creation and initialization would be really nice.

File: apps/api/src/app/widgets/e2e/get-subscriber-preference.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    subscriberToken = token;
+    subscriberProfile = profile;
+  });
+
+  it('should fetch a default user preference', async function () {
+    await session.triggerEvent(template.triggers[0].identifier, subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, subscriberId);
+
+    await session.awaitRunningJobs(template._id);
```
Comment (scopsy) [inline apps/api/src/app/widgets/e2e/get-subscriber-preference.e2e.ts:45]: Why do we need to create triggers here for testing the user preference fetch ?

File: apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@nestjs/common';
+import { SubscriberPreferenceRepository, NotificationTemplateRepository, NotificationTemplateEntity } from '@novu/dal';
+import { GetSubscriberPreferenceCommand } from './get-subscriber-preference.command';
+
+@Injectable()
+export class GetSubscriberPreference {
+  private notificationTemplateRepository = new NotificationTemplateRepository();
```
Comment (scopsy) [inline apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts:7]: why not have it in the constructor here as always and use nestjs DI (dependency injection)?

File: apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+       * add here the template(or fallback) is critical as well
+       */
+
+      return getNoSettingFallback(template);
+    });
+
+    /*
+     * check what param the client need
+     * and create response interface
+     * remove useless params on repo query
```
Comment (scopsy) [inline apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts:46]: Is this a TODO: comment or just comment explaining something? Suggest adding a TODO if this is something you will need to do in the future.

File: apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+     */
+
+    return preferences;
+  }
+}
+
+function getNoSettingFallback(template: NotificationTemplateEntity) {
```
Comment (scopsy) [inline apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts:53]: Would suggest creating an interface for this aswell so later we can specify the response DTO for the API aswell.

File: apps/api/src/app/widgets/widgets.controller.ts [hunk 8]
```diff
@@ -143,6 +146,18 @@ export class WidgetsController {
     return await this.getOrganizationUsecase.execute(command);
   }
 
+  @UseGuards(AuthGuard('subscriberJwt'))
+  @Get('/subscriber-preference')
```
Comment (scopsy) [inline apps/api/src/app/widgets/widgets.controller.ts:150]:
```suggestion
  @Get('/preferences')
```

File: libs/dal/src/repositories/subscriber-preference/subscriber-preference.repository.ts (hunk 1 above)
Comment (scopsy) [reply libs/dal/src/repositories/subscriber-preference/subscriber-preference.repository.ts:10]: I think it's not fetching templates here but subscriber preference. So `findSubscriberPreferences` would be great. Tend to agree with Gali

Comment (djabarovgeorge) [reply libs/dal/src/repositories/subscriber-preference/subscriber-preference.repository.ts:10]: yea it's the preference, awesome thanks for the feedback! :)

File: apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts (hunk 5 above)
Comment (djabarovgeorge) [reply apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts:7]: totally right meant it

File: apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts (hunk 6 above)
Comment (djabarovgeorge) [reply apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts:46]: Its personal todo for the cycle so nothing more then that.

File: apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts (hunk 7 above)
Comment (djabarovgeorge) [reply apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts:53]: Thanks for the heads up i already did on the next branch 🔥

File: apps/api/src/app/widgets/e2e/get-subscriber-preference.e2e.ts (hunk 3 above)
Comment (djabarovgeorge) [reply apps/api/src/app/widgets/e2e/get-subscriber-preference.e2e.ts:51]: Love it! 
Done! ♥️

**Pull request merged**